### ISSUE TITLE
make busted?

### ISSUE THREAD

**Issue by joejoseph00** [wet-boew/wet-boew-drupal#247]
I'm getting this error:

Unable to untar /tmp/make_tmp_1345668033_503543c1ac89b/json2.js. [48.69 sec, 12.33 MB] 

has anyone else done a build lately?

**Comment by sylus** [comment]
The build is under Continuous Integration see: http://travis-ci.org/#!/wet-boew/wet-boew-drupal and currently is green :)

Likely this could be due to an invalid cache... Trying flush your drush caches: "drush cache-clear" and running the make again. Also can try to delete manually your drush cache directory.

If this still doesn't work post back here :)

**Comment by patcon** [comment]
Might suggest linked to a gist with the `drush make --debug` output too :)
Also, drush version?

http://gist.github.com

**Comment by eleclerc** [comment]
I've seen the same problem last week, make sure you're using Drush 5.7 or a 5.6 with this patch http://drupal.org/node/1724108

**Comment by eleclerc** [comment]
we've investigated, and drush 5.7 is broken too, we've had to revert to drush 5.4 to be able to build the distro. thought that others might like to know that before wasting time investigating :)

**Comment by sylus** [comment]
Hmm this is weird. I am using drush 5.4 myself but Travis is using drush 5.6. Would anyone be able to post a drush debug output. I want to update to drush 5.7 as it supports automatically downloading the translation files for each project but before do so would love to make sure everything works.

This all is another reason to hurry up and get our automatic portable VM built up :) see: http://github.com/sylus/wet-boew-drupal-devops

**Comment by joejoseph00** [comment]
The problem is that Drush 5.6 calls a function that doesn't exist in php 5.2 .  We are unable to upgrade to php 5.3 at the moment so we'd like to stick to Drush 5.4 as it works with our current stack

**Comment by sylus** [comment]
Makes sense @joejoseph00 when do you think you can update to php5.3. 

Everyone should really be using php5.3 at the moment for both the security and performance (up to 15% faster just be the mere virtue of using it) improvements.

This is especially true now that php5.4 is extremely stable and offers even more performance improvement :)

**Comment by joejoseph00** [comment]
I'd switch to 5.3 today if it was my choice however we're at the mercy of the keymasters.  And every time I talk to the keymasters my blood pressure rises to the point where my eyeballs start popping out of their sockets.

**Comment by eleclerc** [comment]
there is two issues here, the drupal-org.make is trying to get the github html page for json2.js module instead of the raw json2.js file (I've just made a pull request). I might be wrong though, please double check my PR. 

Our second issue, as stated by Joseph is that we're using 5.2 so the fileinfo extension is not built in php, so we have to rely on the drush code to figure out the file type, which is buggy since drush 5.5 or 5.6.

(5.2 because the agency runs on SuSE Linux Enterprise Edition 11 SP1. Unfortunately we don't have control over this. yep, I know php5.2 have been end of life'd since July 2010... I've spend the last 3 years working on 5.3, it's kind of painful to go back to 5.2 hehe. just mentioning this are we're probably not the only agency running outdated software)

**Comment by sylus** [comment]
Lol I understand. Ideally I'd like to update to drush 5.7 in the scripts by September 17th before the release. Do you think that timeline would be enough? :)

Other departments are a concern but once we give Shared Serices our preconfigure vagrant server that exports to vmdk it should be simple for us all to have the same stack? Do you guys get your servers from SSC?

**Comment by joejoseph00** [comment]
We do not yet get our servers from SSC.

Our servers are Novell SLES 11 SP1.  We've made requests to get SP2 but we have to be patient.  All other apps have to be tested against it first, and then once it's tested we'll roll it out to all our servers.  However this will take time and we only have a small team here but lots of apps and lots of servers.

**Comment by eleclerc** [comment]
here's the relevant part of the debug trace

`json2 downloaded from https://github.com/douglascrockford/JSON-js/blob/master/json2.js. [30.71 sec, 12.43   [ok]
MB]
No fileinfo or mime_magic support available. [30.71 sec, 12.43 MB]                                              [debug]
Examining /tmp/make_tmp_1345733784_503644984bd14/json2.js headers. [30.72 sec, 12.43 MB]                        [debug]
Unable to determine mime type from header bytes 0x2020 of /tmp/make_tmp_1345733784_503644984bd14/json2.js.     [notice]
[30.72 sec, 12.43 MB]
Examining /tmp/make_tmp_1345733784_503644984bd14/json2.js extension. [30.72 sec, 12.44 MB]                      [debug]
Mime type for json2.js is application/x-tar [30.72 sec, 12.44 MB]                                              [notice]
No fileinfo or mime_magic support available. [30.72 sec, 12.44 MB]                                              [debug]
Examining /tmp/make_tmp_1345733784_503644984bd14/json2.js headers. [30.72 sec, 12.44 MB]                        [debug]
Unable to determine mime type from header bytes 0x2020 of /tmp/make_tmp_1345733784_503644984bd14/json2.js.     [notice]
[30.72 sec, 12.44 MB]
Examining /tmp/make_tmp_1345733784_503644984bd14/json2.js extension. [30.72 sec, 12.44 MB]                      [debug]
Mime type for json2.js is application/x-tar [30.72 sec, 12.45 MB]                                              [notice]
Unable to untar /tmp/make_tmp_1345733784_503644984bd14/json2.js. [30.73 sec, 12.64 MB]`

It's a drush issue, thinking that a .js file is a tar file, I'll try to find time to fill a proper bug report on d.o

btw, we're not lucky enough to be allowed to install any virtual machine here

**Comment by eleclerc** [comment]
btw, by reverting to drush-5.4 we can still continue working without issue.

`pear uninstall drush/drush
pear install drush/drush-5.4`

**Comment by eleclerc** [comment]
IMO, this issue can probably be closed as it's really specific to our setup of drush-5.6/php5.2.14/SLES11SP1 and it's related to an unsupported version of PHP anyway :)

plus, we have a workaround for now while we wait for php5.3

**Comment by patcon** [comment]
Can we reopen this to discuss whether we want to make any guarantees about working for specific php and/or drush versions? For example, drush 5.5 and 5.6 seem a little unstable, but 5.4 and 5.7 would seem to be a good bet. (5.4 also runs on acquia, I believe).

travis ci allow testing on multiple versions of php natively, so that's simple. For testing different drush versions, we can make use of travis-ci's ability to do multiple runs with different ENV vars. So if we set a `DRUSH_VERSION` env var as, for example, `5.6` and `5.7`, we could have a test script that runs `pear install drush/drush-${DRUSH_VERSION}.0`. 

So to clarify, the `.travis-ci.yml` would have something like this:

``` yaml
php:
  - 5.2
  - 5.3

env:
  DRUSH_VERSION:
    - 5.6
    - 5.7
```

And this will run for each of:
- php 5.2 running drush 5.6
- php 5.3 running drush 5.6
- php 5.2 running dursh 5.7
- php 5.3 running drush 5.7

**Comment by patcon** [comment]
Also @eleclerc, there's nothing to say that you can't build the site anywhere else (local machine, or in a local VM) and just rsync (or capistrano, if you use that) it to the production/staging server :)

**Comment by eleclerc** [comment]
Hi @patcon, I hear you. in my real life this I can develop locally on osx, and test in ubuntu and centos in virtualbox... but from 9 to 5, as a public servant, I have to play by the network security policies, which means I can't control the server env, and I can't install any software on my local machine, no virtualbox, no apache, no ruby or php... I'm new at GoC, but I assumed the same set of policies applies for everybody. This is why I raised this as an issue, if the WET distro is targeting GoC, and the GoC developers are not allowed to build the distro, that's kinda weird.

oh, and what about building the distro at home and bringing the result on a usb key? that would work, but I'd be in trouble as our policies explicitly says that usb keys are forbidden too :S

In summary, the 5 to 9 me don't think that this is an issue, as I've been managing servers since 1998 and I can setup a proper working environment and every developer should know how to setup their own environment too. But the 9 to 5 me (the public servant) can't build the distro because of the imposed restrictions... anyway, reverting to drush 5.4 did the job for us

**Comment by eleclerc** [comment]
@patcon, just to make sure as English is not my first language.. if my last answer sounds rude,  that's totally unintentional as I 100% agree with you ;-)

**Comment by sylus** [comment]
Yeah this is where vagrant and virtual box comes in which I think we can get shared services to validate our stack. Or at the very least convert to vmdk where the it folks can easily spool it up. More then likely some policies will have to be adapted. Or for testing we could rely on amazon ec2 but would probably go against Canadian soil.

**Comment by joejoseph00** [comment]
I used to like always being on the leading edge, however there is a reason why a lot of people also call this the "bleeding edge".  There is a fine line to walk there.  Amazingly the government seems to be pushing leaps and bounds ahead from where we were a year ago with CLF 2.0 and Interwoven.  I've been burned on development projects where I've totally missed deadlines because I decided to A) decided to upgrade development tools against managements orders and migrate source code into newer compilers that caused new bugs when the older version was stable , (I got let go for that one)  B) go with newer versions of libraries because of decision A, only to find serious bugs with them.  It is 2012, and we've come a long way already.  Staying a couple years behind the curve isn't actually going to restrict you very much as in 2010 /2009 we also had very powerful tools.  I now take a more pragmatic approach, a bit more cautious.  As Dries Buytheart says, the elephants are comming.  We work for the elephants.  And who pulls the elephants strings?  Old people that are near retirement age, much older than any of us.  So we are the mice that can't be seen by the elephants.  Ever see an elephant running scared?  Wouldn't want to be in it's way.  But ya, I'm usually against FUD, but we seriously have to buckle down and chase down real bugs instead of inventing new ones.  I want to see this project work in both languages properly, we haven't even fully put it through it's paces yet.  I'm trying to rally Health Canada into testing the wet distro, I do not want to see this project turn out like our D6 project where the administration was broken in french.  I want quality, and not necessarily all the latest widgets, although ya, go hard if you can do everything, but for me, I want to do the minimum features that I can get away with with the maximum amount of testing, bug fixing and quality.  Changing underwear versions without a business case is asking for trouble.  Keep it simple, keep it working, test , test, test, regression test all new features making sure it doesn't break everything else.  If we want this thing to go global and get international collaboration, even national collaboration, it has to be fully bilingual, and that is the greatest single challenge in Drupal implementations.  We need more resources, more testers, more people implementing, and some sort of regression testing.  Quality means we need at least two people testing every single function in two different ways if possible.
